ols/app/endpoints/ols.py

```diff
@@ -115,6 +113,18 @@ def conversation_request(llm_request: LLMRequest) -> LLMResponse:
     return llm_response
 
 
+def retrieve_conversation_id(llm_request: LLMRequest) -> str:
```

Comment:
@tisnik this function needs to return a second value to indicate whether the conversation id was found in the request, or generated new.  Because we don't want to attempt a conversation cache lookup if we know the conversation id was just generated.  We won't find a conversation if it's a new id and:

1) we don't want to waste the effort on making a request we know will not find anything
2) we don't want to conflate "expected conversation not found in the cache"(merits a warning because while it can happen, it means we've lost history and the user may need to know about it) with "we didn't find it in the cache because it's a new conversation" (not a concern at all, totally expected)

Reply:
Hmm, so with this, it might make sense to do the id generation in the request model. If it would have a field like `new_conversation: bool`. WDYT?

Reply:
meaning all request objects would have a conversationid, and then new_conversation would indicate whether we should expect to find it in the cache or not?

it certainly could be done that way.  i'm not seeing it as a big improvement over "if the request object has no conversation id, it's a new conversation" though.